From 2936604b61e80ef08967e5e2640c8d32f29f03cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Fri, 15 Sep 2023 15:40:50 +0200
Subject: [PATCH] Debug where libLLVM comes from

---
 src/core/rootcling_stage1/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/core/rootcling_stage1/CMakeLists.txt b/src/core/rootcling_stage1/CMakeLists.txt
index a5483b7..1e130ce 100644
--- a/src/core/rootcling_stage1/CMakeLists.txt
+++ b/src/core/rootcling_stage1/CMakeLists.txt
@@ -8,6 +8,8 @@
 # CMakeLists.txt file for building ROOT core/rootcling_stage1 package
 ############################################################################
 
+include(CMakePrintHelpers)
+
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../dictgen/res)
 
 if(WIN32)
@@ -30,6 +32,8 @@ if(NOT builtin_clang)
   link_directories("${LLVM_LIBRARY_DIR}")
 endif()
 
+cmake_print_variables(CLING_LIBRARIES LINK_LIBS CMAKE_DL_LIBS CMAKE_THREAD_LIBS_INIT ROOT_ATOMIC_LIBS)
+
 ROOT_EXECUTABLE(rootcling_stage1 src/rootcling_stage1.cxx
                               $<TARGET_OBJECTS:Clib>
                               $<TARGET_OBJECTS:ClingUtils>
-- 
2.42.0

