From 15b0b07f23b5754eb97fc569c3f0ca5f14cf0a9f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 26 Aug 2024 18:13:29 +0300
Subject: [PATCH] Hardcode include path

---
 .../cling/lib/Interpreter/CIFactory.cpp       | 55 +------------------
 1 file changed, 1 insertion(+), 54 deletions(-)

diff --git a/src/interpreter/cling/lib/Interpreter/CIFactory.cpp b/src/interpreter/cling/lib/Interpreter/CIFactory.cpp
index 2145c11..b659d61 100644
--- a/src/interpreter/cling/lib/Interpreter/CIFactory.cpp
+++ b/src/interpreter/cling/lib/Interpreter/CIFactory.cpp
@@ -115,60 +115,7 @@ namespace {
                                        AdditionalArgList& Args,
                                        const std::string& resourcePath,
                                        bool Verbose) {
-    std::string CppInclQuery("LC_ALL=C ");
-    CppInclQuery.append(Compiler);
-
-    CppInclQuery.append(" -xc++ -E -v /dev/null 2>&1 |"
-                        " sed -n -e '/^.*include/,${' -e '/^ \\/.*include/p' -e '}'");
-
-    if (Verbose)
-      cling::log() << "Looking for C++ headers with:\n  " << CppInclQuery << "\n";
-
-    std::string resourceIncDir;
-    llvm::SmallString<512> recInc(resourcePath);
-    llvm::sys::path::append(recInc, "include");
-    if (llvm::sys::fs::exists(recInc.str()))
-      resourceIncDir = recInc.str().str();
-
-    if (FILE *PF = ::popen(CppInclQuery.c_str(), "r")) {
-      Buf.resize(Buf.capacity_in_bytes());
-      while (fgets(&Buf[0], Buf.capacity_in_bytes(), PF) && Buf[0]) {
-        llvm::StringRef Path(&Buf[0]);
-        // Skip leading and trailing whitespace
-        Path = Path.trim();
-        if (!Path.empty()) {
-          if (!llvm::sys::fs::is_directory(Path)) {
-            if (Verbose)
-              cling::utils::LogNonExistantDirectory(Path);
-          }
-          else {
-            if (!resourceIncDir.empty()) {
-              llvm::SmallString<512> tmp(Path);
-              llvm::sys::path::append(tmp, "xmmintrin.h");
-              if (llvm::sys::fs::exists(tmp.str()))
-                Args.addArgument("-cxx-isystem", resourceIncDir);
-            }
-            Args.addArgument("-cxx-isystem", Path.str());
-          }
-        }
-      }
-      ::pclose(PF);
-    } else {
-      ::perror("popen failure");
-      // Don't be overly verbose, we already printed the command
-      if (!Verbose)
-        cling::errs() << " for '" << CppInclQuery << "'\n";
-    }
-
-    // Return the query in Buf on failure
-    if (Args.empty()) {
-      Buf.resize(0);
-      Buf.insert(Buf.begin(), CppInclQuery.begin(), CppInclQuery.end());
-    } else if (Verbose) {
-      cling::log() << "Found:\n";
-      for (const auto& Arg : Args)
-        cling::log() << "  " << Arg.second << "\n";
-    }
+    Args.addArgument("-cxx-isystem", std::string(::getenv("PREFIX")) + std::string("/include/c++/v1"))
   }
 
   static bool AddCxxPaths(llvm::StringRef PathStr, AdditionalArgList& Args,
-- 
2.46.0

