From 25e19d7a9239286dd2f4336fec33d064f05d73de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 21 Jun 2021 00:37:13 +0200
Subject: [PATCH 1/2] add missing headers in cross comilation

---
 src/core/metacling/src/TCling.cxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/metacling/src/TCling.cxx b/src/core/metacling/src/TCling.cxx
index 38b9e4a..89f8c5e 100644
--- a/src/core/metacling/src/TCling.cxx
+++ b/src/core/metacling/src/TCling.cxx
@@ -17,6 +17,9 @@ Cling is a full ANSI compliant C++-11 interpreter based on
 clang/LLVM technology.
 */
 
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
+
 #include "TCling.h"
 
 #include "ROOT/FoundationUtils.hxx"
-- 
2.32.0

