From c84e108727928cfee76ebc5f9931b7b02a6e8d2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 21 Jun 2021 15:25:53 +0200
Subject: [PATCH] use pre-built dictionary and rootmap

---
 src/cmake/modules/RootMacros.cmake | 34 ++++++++++--------------------
 1 file changed, 11 insertions(+), 23 deletions(-)

diff --git a/src/cmake/modules/RootMacros.cmake b/src/cmake/modules/RootMacros.cmake
index 3f7d9ee..7f586e2 100644
--- a/src/cmake/modules/RootMacros.cmake
+++ b/src/cmake/modules/RootMacros.cmake
@@ -479,16 +479,14 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
   endif()
 
   #---call rootcint------------------------------------------
-  add_custom_command(OUTPUT ${dictionary}.cxx ${pcm_name} ${rootmap_name} ${cpp_module_file}
-                     COMMAND ${command} -v2 -f  ${dictionary}.cxx ${newargs} ${excludepathsargs} ${rootmapargs}
-                                        ${definitions} "$<$<BOOL:${module_defs}>:-D$<JOIN:${module_defs},;-D>>"
-                                        ${includedirs} "$<$<BOOL:${module_incs}>:-I$<JOIN:${module_incs},;-I>>"
-                                        ${ARG_OPTIONS} ${headerfiles} ${_linkdef}
-                     IMPLICIT_DEPENDS ${_implicitdeps}
-                     DEPENDS ${_list_of_header_dependencies} ${_linkdef} ${ROOTCINTDEP}
-                             ${MODULE_LIB_DEPENDENCY} ${ARG_EXTRA_DEPENDENCIES}
-                             ${runtime_cxxmodule_dependencies}
-                     COMMAND_EXPAND_LISTS)
+  add_custom_command(OUTPUT ${dictionary}.cxx
+                     COMMAND ${CMAKE_COMMAND} -E copy
+                     $ENV{RECIPE_DIR}/rootcling/${dictionary}.cxx
+                     ${CMAKE_CURRENT_BINARY_DIR}/${dictionary}.cxx)
+  add_custom_command(OUTPUT ${rootmap_name}
+                     COMMAND ${CMAKE_COMMAND} -E copy
+                     $ENV{RECIPE_DIR}/rootcling/${rootmap_name}
+                     ${CMAKE_BINARY_DIR}/lib/${rootmap_name})
 
   # If we are adding to an existing target and it's not the dictionary itself,
   # we make an object library and add its output object file as source to the target.
@@ -509,7 +507,7 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     target_include_directories(${dictionary} PRIVATE
       ${includedirs} $<TARGET_PROPERTY:${ARG_MODULE},INCLUDE_DIRECTORIES>)
   else()
-    add_custom_target(${dictionary} DEPENDS ${dictionary}.cxx ${pcm_name} ${rootmap_name} ${cpp_module_file})
+    add_custom_target(${dictionary} DEPENDS ${dictionary}.cxx ${rootmap_name})
   endif()
 
   if(PROJECT_NAME STREQUAL "ROOT")
@@ -528,19 +526,9 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
 
   if(NOT ARG_NOINSTALL AND NOT CMAKE_ROOTTEST_DICT AND DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
     ROOT_GET_INSTALL_DIR(shared_lib_install_dir)
-    # Install the C++ module if we generated one.
-    if (cpp_module_file)
-      install(FILES ${cpp_module_file}
-                    DESTINATION ${shared_lib_install_dir} COMPONENT libraries)
-    endif()
 
-    if(ARG_STAGE1)
-      install(FILES ${rootmap_name}
-                    DESTINATION ${shared_lib_install_dir} COMPONENT libraries)
-    else()
-      install(FILES ${pcm_name} ${rootmap_name}
-                    DESTINATION ${shared_lib_install_dir} COMPONENT libraries)
-    endif()
+    install(FILES ${rootmap_name}
+                  DESTINATION ${shared_lib_install_dir} COMPONENT libraries)
   endif()
 
   if(ARG_BUILTINS)
-- 
2.32.0

