From b7557cb75d3f3213ffb024aa3793ec134b901209 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 21 Jun 2021 23:03:50 +0200
Subject: [PATCH] add ThreadLocalStorage.h directory

---
 src/core/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 42eefab..dcc81e4 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -55,6 +55,7 @@ include_directories(
   meta/inc
   gui/inc
   textinput/inc
+  thread/inc
 )
 
 set(objectlibs $<TARGET_OBJECTS:Base>
-- 
2.32.0

