From 27ee4c04891c8e458ce4aae50612f43d232f127d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 22 Jul 2025 18:40:57 +1100
Subject: [PATCH 7/7] don't try to rebuild vendored cling

---
 src/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3c1dac3..b0ea389 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -239,7 +239,6 @@ foreach(artifact_file ${artifact_files})
 endforeach()
 add_custom_target(move_artifacts DEPENDS ${stamp_file} ${artifact_files_builddir})
 
-add_subdirectory (interpreter)
 # Add the compilation flags from LLVM to the flags we use to compile cling-based tools.
 set(CLING_CXXFLAGS "${CLING_CXXFLAGS} ${ROOT_LLVM_FLAGS}")
 
